aiosmtpd/testing/helpers.py

```diff
@@ -82,3 +84,19 @@ def assert_auth_required(testcase: TestCase, code, response):
 )
 
 SUPPORTED_COMMANDS_NOTLS = SUPPORTED_COMMANDS_TLS.replace(b" STARTTLS", b"")
+
+
+def send_recv(
+        sock: socket.socket, data: bytes, end: bytes = b"\r\n", timeout=0.1
+) -> bytes:
+    sock.send(data + end)
+    slist = [sock]
+    result: List[bytes] = []
+    while True:
+        read_s, _, _ = select.select(slist, [], [], timeout)
+        if read_s:
+            # We can use sock instead of read_s because slist only contains sock
+            result.append(sock.recv(1024))
+        else:
+            break
+    return b"".join(result)
```

Comment:
I need this func to perform low-level non-hanging send-and-receive for each message line during `DATA` phase.